src/components/Autocomplete/Autocomplete.jsx

```diff
@@ -0,0 +1,291 @@
+import React from 'react';
+import _ from 'lodash';
+import { lucidClassNames } from '../../util/style-helpers';
+import { createLucidComponentDefinition } from '../../util/component-definition';
+import * as reducers from './Autocomplete.reducers';
+import * as KEYCODE from '../../constants/key-code';
+import DropMenu from '../DropMenu/DropMenu';
+import CaretIcon from '../Icon/CaretIcon/CaretIcon';
+
+const boundClassNames = lucidClassNames.bind('&-Autocomplete');
+
+const {
+	arrayOf,
+	bool,
+	func,
+	object,
+	shape,
+	string
+} = React.PropTypes;
+
+/**
+ *
+ * {"categories": ["controls", "text"], "madeFrom": ["CaretIcon", "DropMenu"]}
+ *
+ * A text input with suggested values displayed in an attached menu.
+ */
+
+const Autocomplete = React.createClass(createLucidComponentDefinition({
+	displayName: 'Autocomplete',
+
+	reducers,
+
+	propTypes: {
+		/**
+		 * Appended to the component-specific class names set on the root elements.
+		 */
+		className: string,
+		/**
+		 * Styles that are passed through to root element.
+		 */
+		style: object,
+		/**
+		 * Disables the Autocomplete from being clicked or focused.
+		 */
+		isDisabled: bool,
+		/**
+		 * Array of suggested text input values shown in drop menu.
+		 */
+		suggestions: arrayOf(string),
+		/**
+		 * Text value of the input.
+		 */
+		value: string,
+		/**
+		 * Object of DropMenu props which are passed thru to the underlying DropMenu component.
+		 */
+		DropMenu: shape(DropMenu.propTypes),
+		/**
+		 * Called when the input value changes.
+		 * Has the signature `(value, {props, event}) => {}` where value is a string.
+		 */
+		onChange: func,
+		/**
+		 * Called when a suggestion is selected from the menu.
+		 * Has the signature `(optionIndex, {props, event}) => {}` where optionIndex is a number.
+		 */
+		onSelect: func,
+		/**
+		 * Called when menu is expected to expand.
+		 * Has the signature `({props, event}) => {}`.
+		 */
+		onExpand: func,
+	},
+
+	getDefaultProps() {
+		return {
+			isDisabled: false,
+			suggestions: [],
+			value: '',
+			onChange: _.noop,
+			onSelect: _.noop,
+			onExpand: _.noop,
+			DropMenu: DropMenu.getDefaultProps()
+		};
+	},
+
+	handleSelect(optionIndex, { event }) {
+		const {
+			suggestions,
+			onChange,
+			onSelect,
+		} = this.props;
+
+		onChange(suggestions[optionIndex], { event, props: this.props});
+		onSelect(optionIndex, { event, props: this.props});
+	},
+
+	handleInput(event) {
+		const {
+			onChange,
+			onExpand,
+			DropMenu: {
+				onCollapse
+			}
+		} = this.props;
+
+		onChange(event.target.value, {event, props: this.props});
+		if (!_.isEmpty(event.target.value)) {
+			onExpand({event, props: this.props});
+		} else {
+			onCollapse();
+		}
+	},
+
+	getInputValue() {
+		return _.get(this.refs, 'inputNode.value', this.props.value);
+	},
+
+	setInputValue(value) {
+		if (this.refs.inputNode) {
+			this.refs.inputNode.value = value;
```

Comment:
Really only if we're still undecided as to whether to stick with `ref='foo'` or go with `ref={function () { ... }}`...